core/src/main/resources/jenkins/model/BuildDiscarderProperty/help.html

```diff
@@ -1,19 +1,51 @@
 <div>
-  This controls the disk consumption of Jenkins by managing how long you'd like to keep
-  records of the builds (such as console output, build artifacts, and so on.)
-  Jenkins offers two criteria:
-
+  This determines when, if ever, build records for this project should be
+  discarded. Build records include the console output, archived artifacts, and
+  any other metadata related to a particular build.
+  <p>
+  Keeping fewer builds means less disk space will be used in the <i>Build Record
+  Root Directory</i>, which is specified on the <i>Configure System</i> screen.
+  <p>
+  <!-- TODO: Most/all of the following should be dynamic, showing info from the
+       `BuildDiscarder` implementations that are currently loaded -->
+  Jenkins offers two options for determining when builds should be discarded:
   <ol>
     <li>
-      Driven by age. You can have Jenkins delete a record if it reaches a certain age
-      (for example, 7 days old.)
+      Build age: discard builds if they reach a certain age; for example, seven
+      days old.
     <li>
-      Driven by number. You can have Jenkins make sure that it only maintains up to
-      N build records. If a new build is started, the oldest record will
-      be simply removed.
+      Build count: discard the oldest build if a certain number of builds
+      already exist.
   </ol>
+  These two options can be active at the same time, so you can keep builds for
+  14 days, but only up to a limit of 50 builds, for example.  If either limit is
+  exceeded, then any builds beyond that limit will be discarded.
+  <p>
+  You can also ensure that important builds are kept forever, regardless of the
+  setting here &mdash; click the <i>Keep this build forever</i> button on the
+  build page.
+  <br>
```

Comment:
The thing is, if we kind of define and follow some semantic things like those, we could very probably even if necessary define some specific classes to be used inside those HTML doc snippets, and make them visually consistent.

**(NOTE: thanks very much anyway & again, in case it's not obvious. What you did there is in any case a gigantic improvement of what was previously present).**